frb/galaxies/galfit.py

```diff
@@ -0,0 +1,480 @@
+                    # TODO: accommodate more complex fits.
+                    instance.append((item, pp, lines[kk:kk + 10][pp + 1]))  # and keep the model and error
+
+    # only keep the latest one in case there are multiple runs
+    fit_out, line, err_out = instance[-1]
+
+    # Use regex to identify all floats in fit_out and err_out
+    float_regex = r"\d+\.\d+" # Any string of the form <int>.<int> (which is just a float)
+
+    # Find all instances of float in the strings and convert from string to float
+    fitparams = np.array(re.findall(float_regex, fit_out)).astype('float')
+    fiterrs = np.array(re.findall(float_regex, err_out)).astype('float')
+
+    #Store values
+    pix_dict = {'x':fitparams[0], 'y':fitparams[1],
+              'mag':fitparams[2], 'reff':fitparams[3],
+              'n':fitparams[4], 'b/a':fitparams[5], 'PA':fitparams[6],
+              'x_err':fiterrs[0], 'y_err':fiterrs[1],
+              'mag_err':fiterrs[2], 'reff_err':fiterrs[3],
+              'n_err':fiterrs[4], 'b/a_err':fiterrs[5], 'PA_err':fiterrs[6],
+              }
+
+    return pix_dict
+
+def pix2coord(pix_dict:dict, wcs:WCS, table:Table=False)->dict:
+    """
+    Takes the output table from galfit's
+    fit.log file and converts all pixel
+    measurements to physical measurements.
+    Args:
+        pix_dict (dict): Raw sersic
+            fit parameter dict from fit.log
+        wcs (WCS): Input image WCS.
+        table (bool, optional): Return as
+            a table instead?
+    Returns:
+        sky_dict (dict/Table): pix_dict
+            translated to angular units
+            on the sky.
+    """
+
+    sky_dict = {}
+    platescale = get_platescale(wcs)
+
+    # Centroids
+    xpix, ypix = pix_dict['x']-1, pix_dict['y']-1
+    centr_coords = wcs.pixel_to_world(xpix, ypix)
+    sky_dict['ra'] = centr_coords.ra.value
+    sky_dict['dec'] = centr_coords.dec.value
+    sky_dict['ra_err'] = pix_dict['x_err']*platescale
+    sky_dict['dec_err'] = pix_dict['y_err']*platescale
+    # Magnitude
+    sky_dict['mag'] = pix_dict['mag']
+    sky_dict['mag_err'] = pix_dict['mag_err']
+    # Half-light radius
+    sky_dict['reff_ang'] = pix_dict['reff']*platescale
+    sky_dict['reff_ang_err'] = pix_dict['reff_err']*platescale
+    # Sersic index
+    sky_dict['n'] = pix_dict['n']
+    sky_dict['n_err'] = pix_dict['n_err']
+    # Axis ratio
+    sky_dict['b/a'] = pix_dict['b/a']
+    sky_dict['b/a_err'] = pix_dict['b/a_err']
+    # Sky position angle
+    sky_dict['PA'] = pix_dict['PA']
+    sky_dict['PA_err'] = pix_dict['PA_err']
+
+    if table:
+        sky_dict = Table([sky_dict])
+        # Reorder because dict-> Table messes it up
+        sky_dict = sky_dict['ra','ra_err','dec','dec_err','mag',
+                            'mag_err','reff_ang','reff_ang_err',
+                            'n','n_err','b/a','b/a_err','PA','PA_err']
+    return sky_dict
+
+def run(imgfile:str, psffile:str, **kwargs)->int:
```

Comment:
Returns an `int` or `str`?

Reply:
Int. It is a fit outcome flag. Corrected now.